presto-main/src/main/java/com/facebook/presto/sql/planner/plan/Assignments.java

```diff
@@ -164,6 +164,27 @@ public int size()
         return assignments.size();
     }
 
+    @Override
+    public boolean equals(Object o)
```

Comment:
Making a class implement equals/hashCode is profound decision in that it is irreversible, as it's very hard to track who came to depend on this.

Usefulness of equality of `Assignments` is hindered by the fact equality on `Expression`-s has limited usefulness either. Is `a + 1` equal to `a + 1`? It is. Are the expressions equivalent? It depends...

Since `Assignments` expose their `assignments`, you are free to use `Equivalence`. Please consider.